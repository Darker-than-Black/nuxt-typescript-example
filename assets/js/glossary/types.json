{
  "String": "string",
  "Array": "array",
  "Number": "number",
  "Object": "object"
}
