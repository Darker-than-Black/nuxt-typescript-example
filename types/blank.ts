export interface BlankItemField {
  title: string,
  path: string,
  itemClass?: string,
  countryPath?: string
}

export interface BlankItemList extends BlankItemField {
  items: string[]
}

export interface BlankItemConfig {
  type: string,
  path: string,
  title?: string,
  countryPath?: string,
  width?: string,
  class?: string,
  headerClass?: string,
  itemsContentClass?: string,
  itemListClass?: string,
  items: BlankItemField[],
  icons?: BlankItemList[]
}
