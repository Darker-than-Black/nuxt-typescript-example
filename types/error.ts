export type HttpError = {
  field: string,
  message: string
}

export type ErrorHandler = {
  statuses: Array<number>,
  return: any
}

export type ErrorConfig = {
  returnValue?: any,
  errorsList?: Array<ErrorHandler>
}

export type ErrorCode = {
  error?: number
}
