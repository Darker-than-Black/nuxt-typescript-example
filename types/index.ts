export * from './input'
export * from './router'
export * from './error'
export * from './table'
export * from './link'
export * from './blank'

// api
export * from './api/declaration/declarations'
export * from './api/declaration/declaration'
export * from './api/car-goods'
export * from './api/entry-journal/entry-journal'
export * from './api/entry-journal/entry-journal-ticket'
export * from './api/dictionary'

// UI
export * from './UI'

// tests
export * from './unit-test'
