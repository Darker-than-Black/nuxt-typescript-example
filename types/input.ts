export type InputValue = number|string|boolean

export type MultiSelectValue = InputValue[]

export type SelectOptions = SelectOption[]

export type SelectOption = {
  value: InputValue,
  text: string
}

export type TableRowValue = {
  [key: string]: FormFieldValue
}

export type FormFieldValue = InputValue|MultiSelectValue[]|TableRowValue[]|object

export interface IFormFieldComponentProps {
  value?: FormFieldValue,
  readonly label?: string,
  readonly disabled?: boolean,
  readonly inputClasses?: string[]|string,
  // input attrs
  readonly type?: string,
  readonly placeholder?: string,
  // select attrs
  readonly options?: SelectOptions,
  readonly filterKey?: string,
  // for validate
  readonly vid?: string,
  readonly rules?: string,
  // for data list
  readonly listLink?: string,
}

export interface IFactoryFieldConfig extends IFormFieldComponentProps {
  // ~/assets/js/glossary/field-types.json
  readonly fieldType: string,
  readonly defaultValue: FormFieldValue,
  readonly invisible?: boolean,
  readonly fields?: {
    [key: string]: IFactoryFieldConfig
  }
}

export interface IFactoryFieldsConfig {
  [key: string]: IFactoryFieldConfig
}
