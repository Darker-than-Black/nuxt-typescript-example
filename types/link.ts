export type Links = {
  [key: string]: Link
}

export type Link = {
  text: string,
  icon?: string,
  path: string,
  subLinks?: Links
}
