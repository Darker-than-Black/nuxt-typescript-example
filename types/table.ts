export type TableConfig = {
  size?: number,
  totalElements?: number,
  totalPages?: number,
  number?: number // page number
}

// Doc: https://bootstrap-vue.org/docs/components/table#field-definition-reference
export type TableColConfig = {
  key: string,
  label: string,
  sortable: boolean,
  tdClass: string[] | string,
  thClass: string[] | string
}

export type TableSortEvent = {
  apiUrl: string,
  currentPage: number,
  filter: any
  perPage: number,
  sortBy: string
  sortDesc: boolean
}
