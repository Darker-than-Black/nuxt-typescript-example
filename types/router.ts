import { InputValue, MultiSelectValue } from '~/types/input'

export type QueryParam = MultiSelectValue | InputValue

export type QueryParams = {
  [key: string]: QueryParam
}

export type QueryParamsConfig = {
  [key: string]: {
    defaultValue: QueryParam,
    type: string
  }
}
