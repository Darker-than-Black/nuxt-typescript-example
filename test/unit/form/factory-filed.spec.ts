import VueTestInstance from '~/test/factory/vue-instance'
import FactoryField from '~/components/form/factory-field.vue'
import CONFIG from '~/assets/js/glossary/form/create-ticker'
import ConvertFieldsConfigToDefaultValue from '~/assets/js/utils/convert-fields-config-to-default-value'
import { CreateTicket } from '~/types'

const dataDefaultValue: CreateTicket = ConvertFieldsConfigToDefaultValue(CONFIG) as CreateTicket
describe('factory-field.vue', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = VueTestInstance(FactoryField, {
      options: {
        propsData: {
          config: CONFIG.documents,
          value: dataDefaultValue.documents
        }
      }
    })
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check emit: input()', () => {
    expect(wrapper.vm.input(dataDefaultValue.documents)).toEqual(dataDefaultValue.documents)
  })
})
