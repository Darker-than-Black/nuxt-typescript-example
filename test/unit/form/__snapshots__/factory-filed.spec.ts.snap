// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`factory-field.vue should match the snapshot 1`] = `
<div>
  <management-table-stub value="[object Object]" label="" placeholder="" type="text" inputclasses="" vid="" rules="" listlink="" config="[object Object]" btntext="Додати документ"></management-table-stub>
</div>
`;
